src/__tests__/vue-apollo.js

```diff
@@ -0,0 +1,55 @@
+import '@testing-library/jest-dom'
+import {render, fireEvent, screen, waitFor} from '@testing-library/vue'
+import VueApollo from 'vue-apollo'
+
+// Since vue-apollo doesn't provides a MockProvider for Vue,
+// you need to use some kind of a mocking client for Apollo.
+
+// If you decide to use `mock-apollo-client`,
+// have a look at the documentation at
+// https://github.com/Mike-Gibson/mock-apollo-client
+import {createMockClient} from 'mock-apollo-client'
+
+import Component from './components/VueApollo.vue'
+import {userQuery, updateUserMutation} from './components/VueApollo/queries'
+
+test('mocking queries and mutations', async () => {
+  const mockClient = createMockClient()
+
+  mockClient.setRequestHandler(userQuery, () =>
+    Promise.resolve({
+      data: {user: {id: '1', email: 'alice@example.com'}},
+    }),
+  )
+
+  render(Component, {props: {id: '1'}}, localVue => {
+    localVue.use(VueApollo)
+
+    return {
+      apolloProvider: new VueApollo({
+        defaultClient: mockClient,
+      }),
+    }
+  })
+
+  await waitFor(() => screen.getByText('alice@example.com'))
+  expect(screen.getByText('alice@example.com')).toBeInTheDocument()
+
+  await fireEvent.update(
+    screen.getByLabelText('Email'),
+    'alice+new@example.com',
+  )
+
+  mockClient.setRequestHandler(updateUserMutation, variables =>
```

Comment:
quick question: Is it needed to define this request handler right here? I was reading the text and felt a bit off to have some "setting up" stuff between the `fireEvent.update` and the `fireEvent.click`

Reply:
Oh no, totally not, you can define it right in the beginning, next to the query.
This is also highly specific to the mocking client, usually when you come from React's MockProvider you need to define all queries and mutations beforehand.

Reply:
cool 💯! Then I'd suggest moving it to the beginning of the test, so one can easily spot the "Arrange" section of it :)